docs/remote_inference_blueprints/cohere_connector_embedding_blueprint.md

```diff
@@ -0,0 +1,361 @@
+
+Cohere currently offers the following Embedding models (with model name and embedding dimensions). Note that only the following have been tested with the blueprint guide.
+
+- embed-english-v3.0 1024
+- embed-english-v2.0 4096
+
+See [Cohere's /embed API docs](https://docs.cohere.com/reference/embed) for more details.
+
+#### 1; Update your Opensearch cluster settings
+
+```json
+PUT /_cluster/settings
+
+{
+    "persistent": {
+        "plugins.ml_commons.trusted_connector_endpoints_regex": [
+          "^https://api\\.cohere\\ai/.*$",
+        ]
+    }
+}
+```
+
+If you have existing cluster settings, make sure to add them inside the array of endpoints.
+
+#### 2. Create your Model connector and Model group
+
+##### 2a. Register your Model group
+
+```json
+POST /_plugins/_ml/model_groups/_register
+
+{
+  "name": "cohere_model_group",
+  "description": "Your Cohere model group"
+}
+```
+
+This request response will return the `model_group_id`, note it down.
+
+##### 2b. Create your Model connector
+
+See above for all the values the `parameters > model` parameter can take.
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "Cohere Embed Model",
+  "description": "The connector to Cohere's public embed API",
+  "version": "1",
+  "protocol": "http",
+  "credential": {
+    "cohere_key": "<ENTER_COHERE_API_KEY_HERE>"
+  },
+  "parameters": {
+    "model": "<ENTER_MODEL_NAME_HERE>", // Choose a Model from the provided list above
+    "input_type":"search_document",
+    "truncate": "END"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://api.cohere.ai/v1/embed",
+      "headers": {
+        "Authorization": "Bearer ${credential.cohere_key}",
+        "Request-Source": "unspecified:opensearch"
+      },
+      "request_body": "{ \"texts\": ${parameters.texts}, \"truncate\": \"${parameters.truncate}\", \"model\": \"${parameters.model}\", \"input_type\": \"${parameters.input_type}\" }",
+      "pre_process_function": "connector.pre_process.cohere.embedding",
+      "post_process_function": "connector.post_process.cohere.embedding"
+    }
+  ]
+}
+```
+
+This request response will return the `connector_id`, note it down.
+
+##### 2c. Register your Model connector
+
+You will now register the model you created using the `model_group_id` and `connector_id` from the previous requests.
+
+```json
+POST /_plugins/_ml/models/_register
+Content-Type: application/json
+
+{
+    "name": "Cohere Embed Model",
+    "function_name": "remote",
+    "model_group_id": "<MODEL_GROUP_ID>",
+    "description": "Your Cohere Embedding Model",
+    "connector_id": "<CONNECTOR_ID>"
+}
+```
+
+This will create a registration task, the response should look like:
+
+```json
+{
+  "task_id": "9bXpRY0BRil1qhQaUK-u",
+  "status": "CREATED",
+  "model_id": "9rXpRY0BRil1qhQaUK_8"
+}
+```
+
+You can then check whether the registration task has completed with a GET request:
```

Comment:
Line 108 to 127 is not needed, as user can see the model id directly from line 104